### ISSUE TITLE
azapi_resource_action: Microsoft.Web/sites config does not detect changes

### ISSUE THREAD

**Issue by phatcher** [Azure/terraform-provider-azapi#557]
As the app service terraform provider does not yet support setting the minimum TLS cipher suite I had a go at setting it via azapi, something like this

```
data "azapi_resource_id" "config" {
  type      = "Microsoft.Web/sites/config@2023-12-01"
  parent_id = azurerm_app_service.this.id
  name      = "web"
}

resource "azapi_resource_action" "config" {
  type        = "Microsoft.Web/sites/config@2023-12-01"
  resource_id = data.azapi_resource_id.config.id
  method      = "PATCH"

  body = {
    name = "web"
    properties = {
      minTlsCipherSuite = local.min_tls_ciphersuite
    }
  }
}
```

This works, but if the value is modified via the portal and I re-apply the terraform, no change is applied i.e. it keeps the portal setting.

Is this expected behaviour?

**Comment by stemaMSFT** [comment]
hey @phatcher this may be better implemented through `azapi_update_resource`? Since that is made for this sort of behavior in a CRUD-friendly manner.

**Comment by ms-henglu** [comment]
Yes, the `azapi_resource_action` is only used to trigger an HTTP request, it doesn't monitor the resource's state. If the `minTlsCipherSuite` could be updated by PUT method, it's recommended to use `azapi_update_resource`, because it monitors the state, and will show diff if the value is modified by other client tools.

**Comment by phatcher** [comment]
@stemaMSFT Thanks that is working, there's a schema issue with the VSCode extension as it wants to assign "web" to the name, but the it appears to be the base name of the site.

If I leave it as web it always show a change that never actually sticks

**Comment by stemaMSFT** [comment]
@phatcher are you still running into that issue with the extension? That does sound strange.

**Comment by ms-henglu** [comment]
> @phatcher are you still running into that issue with the extension? That does sound strange.

Hi @stemaMSFT,

I believe this is the issue mentioned by @phatcher :

<img width="1059" alt="image" src="https://github.com/user-attachments/assets/97690796-8efc-446e-a5ed-ed0f5584de20">

I think the cause is bad swagger, the generated document says that the `name` field only supports a list of allowed values: 
<img width="1059" alt="image" src="https://github.com/user-attachments/assets/f61f837a-0f81-4164-8f6f-6cf2e7602946">

**Comment by phatcher** [comment]
What I wasn't sure is whether the schema was correct and the API playing up or vice versa :-)

**Comment by fgarcia-cnb** [comment]
the problem with using 'azapi_update_resource' is that it doesnt use PATCH, it uses PUT, so it replaces the entire appsettings config. 'azapi_resource_action' with methog PATCH does proeprly merge key value pairs with existing values, but does not detect changes

**Comment by benjamin-purcell** [comment]
Hi, this issue was originally reported in 2024, so we’re checking whether it is still relevant. Could you please confirm whether you are still experiencing the problem with the latest available version?

If it is still occurring, please provide any updated details, including the software/provider version, configuration, error messages, and a minimal reproduction if possible. If the issue has been resolved, please let us know what fixed it so we can update our records. Thank you!
